File: demo-vk-rest-scraping.java/src/main/java/demo/model/University.java
package demo.model;

import java.util.List;

public class University {

    private final int id;
    private final String title;

    private Integer facultiesCount;
    private List<Faculty> faculties;

    private University(int id, String title) {
        this.id = id;
        this.title = title;
    }

    public University(ConstructorMediator cm) {
        this(cm.getId(), cm.getTitle());
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public Integer getFacultiesCount() {
        return facultiesCount;
    }

    public List<Faculty> getFaculties() {
        return faculties;
    }

    public void setFaculties(List<Faculty> faculties) {
        this.faculties = faculties;
        this.facultiesCount = faculties.size();
    }

    @Override
    public String toString() {
        return "University{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                '}';
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/model/University.kt
package demo.model

class University private constructor(val id: Int, val title: String) {

    var facultiesCount: Int? = null
        private set

    var faculties: List<Faculty>? = null
        set(faculties) {
            field = faculties
            this.facultiesCount = faculties!!.size
        }

    constructor(cm: ConstructorMediator) : this(cm.id, cm.title) {
    }

    override fun toString(): String {
        return "University{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                '}'
    }
}
